### PR TITLE
Adds partial document updates in data_cosmos

### PR DESCRIPTION
**Pull request by daniel-larsen**
Adds partial document updates in data_cosmos. Partial document updates are described [here](https://learn.microsoft.com/en-us/azure/cosmos-db/partial-document-update) , and the REST API is documented [here](https://learn.microsoft.com/en-us/rest/api/cosmos-db/patch-a-document). An example is shown below. 

```rust
let operations = vec![
     Operation::add("/color", "silver")?,
     Operation::remove("/used"),
     Operation::set("/price", 355.45)?,
     Operation::incr("/inventory/quantity", 10)?,
     Operation::add("/tags/-", "featured-bikes")?,
     Operation::r#move("/inventory/color", "/color"),
];

document_client.patch_document(operations).await?;
```

### CODE REVIEW

File: sdk/data_cosmos/src/operations/patch_document.rs [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+}
+
+#[derive(Serialize, Debug, Clone)]
+#[serde(rename_all = "lowercase")]
+#[serde(tag = "op")]
+pub enum Operation {
+    Add { path: String, value: Value },
+    Remove { path: String },
+    Set { path: String, value: Value },
+    Incr { path: String, value: Value },
```
Comment (demoray) [inline sdk/data_cosmos/src/operations/patch_document.rs:67]: Given the documentation specifies this must be a positive or negative value, what are your thoughts on having this be a `serde_json::value::Number`?

File: sdk/data_cosmos/src/clients/document.rs [hunk 2]
```diff
@@ -42,6 +42,26 @@ impl DocumentClient {
         ReplaceDocumentBuilder::new(self.clone(), document)
     }
 
+    /// Patch the document.
+    /// The Patch Document operation does path-level updates to specific files/properties in a single document.
+    ///
+    /// # Example
+    /// ```no_run
```
Comment (demoray) [inline sdk/data_cosmos/src/clients/document.rs:49]:
```suggestion
    /// ```no_run
    /// # use azure_data_cosmos::prelude::*;
    /// # async fn func(document_client: &DocumentClient) -> azure_core::Result<()> {
```

File: sdk/data_cosmos/src/clients/document.rs [hunk 3]
```diff
@@ -42,6 +42,26 @@ impl DocumentClient {
...
+    ///     Operation::add("/color", "silver")?,
+    ///     Operation::remove("/used"),
+    ///     Operation::set("/price", 355.45)?,
+    ///     Operation::incr("/inventory/quantity", 10)?,
+    ///     Operation::add("/tags/-", "featured-bikes")?,
+    ///     Operation::r#move("/inventory/color", "/color"),
+    /// ];
+    ///    
+    /// document_client.patch_document(operations).await?;
+    /// ```
```
Comment (demoray) [inline sdk/data_cosmos/src/clients/document.rs:60]:
```suggestion
    /// # Ok(())
    /// # }
    /// ```
```

**Review by demoray — APPROVED** [review]

File: sdk/data_cosmos/src/operations/patch_document.rs (hunk 1 above)
Comment (daniel-larsen) [reply sdk/data_cosmos/src/operations/patch_document.rs:67]: That makes sense to me.

**Pull request merged**